### ISSUE TITLE
[UX] Module page: Alphabetize by module name and remove package groupings

### ISSUE THREAD

**Issue by jenlampton** [backdrop/backdrop-issues#3148]
## Describe your issue or idea

This is a follow-up to https://github.com/backdrop/backdrop-issues/issues/37 to un-block that issue.

---
PR by @klonos: https://github.com/backdrop/backdrop/pull/2548

**Comment by klonos** [comment]
FTR, I am ambivalent for this...

- I personally still find package groupings useful
- On the other hand, I also acknowledge the fact that a module might be suitable for inclusion in more than one grouping, and duplication will add confusion to the UX
- Submodules that do not use their "parent" module name in their name, will now need to start doing that, if they wish to be grouped together. Although package grouping might have been (ab)used as was seen fit by the module maintainers over the years, I always saw this feature as a way to group modules of the same "ecosystem" (think views/webform/rules/), without having to necessarily add the name of the "main" module to the name of the rest.
- I recommended adding a package/alfa toggle at some point, but I know that @quicksketch is already skeptical about adding the core/contrib/custom switch (and I find that one more important)

**Comment by klonos** [comment]
I have just filed a PR to get this going. After #37 was implemented, we now have some weird grouping/sorting in this page, so best to remove all grouping and implement #3147 and #3146 

While working on this, I needed to remove the title from the single fieldset used to group all modules. I tried to do that by setting `'#title_display' => 'invisible',`, but that does not seem to work: #3619 

Having filed that bug report, I started wondering if it makes sense to have the module table within a fieldset at all. We don't seem to be doing that in other listing pages.

**Comment by klonos** [comment]
...I would like to get this in 1.13.0 if possible. Added milestone candidate tag.

**Comment by jenlampton** [comment]
We need to solve the UX issue that the groupings/fieldsets solves before
(or when) we remove the groupings. Do you have any recommendations for
that?

**Comment by jenlampton** [comment]
Maybe https://github.com/backdrop/backdrop-issues/issues/3147 would be
enough?

**Comment by klonos** [comment]
I would like to revive this issue 🙂 

@jenlampton yes, I think that #3147 would be enough.

**Comment by klonos** [comment]
Although this is a task and not a feature request, the change is rather substantial, so let please get it in with a minor release. ...ideally, we'll get #3146 and #3147 in the same release.

**Comment by jenlampton** [comment]
All of these issues were blocked by a lack of consensus, and tagged "needs more feedback". Should we escalate PMC? Maybe we can get some direction and have a plan for 1.18 :)

**Comment by klonos** [comment]
Sure, by all means 👍

**Comment by klonos** [comment]
...in the meantime, I have refreshed the PR sandbox for those that would like to test and get a feel of how this would work.

**Comment by stpaultim** [comment]
For the record, I don't think this PR is working as expected right now. 

CSS is wonky and after you type two letters the filter disappears and shows _some_ results?

![Screen Shot 2020-08-30 at 1 28 42 PM](https://user-images.githubusercontent.com/3144571/91666722-c8b4c080-eac4-11ea-9854-b0bafd71bdc9.png)

**Comment by laryn** [comment]
Just seeing this one and I'm thinking for a case like Ubercart, it would introduce a lot of confusion to strip out package groupings. Right now that module comes with multiple package groupings for various sub-modules. I guess one solution for that case would be to do as @klonos suggests, but doesn't seem as clean perhaps:

> Submodules that do not use their "parent" module name in their name, will now need to start doing that, if they wish to be grouped together. Although package grouping might have been (ab)used as was seen fit by the module maintainers over the years, I always saw this feature as a way to group modules of the same "ecosystem" (think views/webform/rules/), without having to necessarily add the name of the "main" module to the name of the rest.

**Comment by docwilmot** [comment]
Search by modules tagged Ubercart?

**Comment by laryn** [comment]
Yes, if we add the word "Ubercart" to all of them, I guess. Unless the intention is to also include tags in searches. e.g.
![Screen Shot 2020-08-30 at 3 16 07 PM](https://user-images.githubusercontent.com/871421/91668638-c9088800-ead3-11ea-81b5-c50e4298d82a.jpg)

**Comment by klonos** [comment]
The Ubercart core and contrib module naming was brought up in Zulip today: https://backdrop.zulipchat.com/#narrow/stream/218635-Backdrop/topic/Ubercart.20module.20titles.20and.20packages

As I mentioned there, if this change here happens, then prepending `Ubercart` to the module names would help:

> ...perhaps `Ubercart: Core: Some Module` or `Ubercart: Category: Some Contrib Module` would be helpful

**Comment by jlfranklin** [comment]
[Module Filter](https://backdropcms.org/project/module_filter) is ported and in contrib.
